hugr-core/src/ops/constant/custom.rs

```diff
@@ -476,4 +476,43 @@ mod test {
             serde_yaml::from_value(serde_yaml::to_value(&ev).unwrap()).unwrap()
         );
     }
+
+    mod proptest {
+        use ::proptest::prelude::*;
+
+        use crate::{
+            extension::ExtensionSet,
+            ops::constant::CustomSerialized,
+            proptest::{any_serde_yaml_value, any_string},
+            types::Type,
+        };
+
+        impl Arbitrary for CustomSerialized {
+            type Parameters = ();
+            type Strategy = BoxedStrategy<Self>;
+            fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+                let typ = any::<Type>();
+                let extensions = any::<ExtensionSet>();
+                // here we manually construct a serialized `dyn CustomConst`.
+                // The "c" and "v" come from the `typetag::serde` annotation on
+                // `trait CustomConst`.
+                // TODO This is not ideal, if we were to randomly
+                // generate a valid tag(e.g. "c" = "ConstInt") then things will
```

Comment:
```suggestion
                // generate a valid tag (e.g. "ConstInt") then things will
```
...in that the `"c" =` is not part of the tag. But the idea is that we are relying on the random `any_string` *not* matching any handler for a recognized type, right?

Perhaps change `randomly` on previous line to `accidentally`, and/or `...then the handler for that type will interpret the "v" and fail`